### PR TITLE
Adding error codes to docker/volume

### PR DESCRIPTION
**Pull request by brahmaroutu**
Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

**Comment by ehazlett** [comment]
@brahmaroutu Would you mind running `gofmt` on this?  Thanks!

**Comment by calavera** [comment]
Those errors have been moved to `volume/store/store.go`, you should move the error definitions there.

**Comment by brahmaroutu** [comment]
thanks @ehazlett @calavera  I will update the PR soon.

**Comment by calavera** [comment]
LGTM

**Comment by duglin** [comment]
@brahmaroutu "TestDaemonRestartRmVolumeInUse" might be a real failure

File: daemon/container_unix.go [hunk 1]
```diff
@@ -1226,7 +1225,7 @@ func (container *Container) removeMountPoints(rm bool) error {
 			// not an error, but an implementation detail.
 			// This prevents docker from logging "ERROR: Volume in use"
 			// where there is another container using the volume.
-			if err != nil && err != store.ErrVolumeInUse {
+			if err != nil && err != derr.ErrorCodeVolumeInUse {
 				rmErrors = append(rmErrors, err.Error())
```
Comment (duglin) [inline daemon/container_unix.go]: sadly, this may need utils.GetErrorMessage()

File: daemon/delete.go [hunk 2]
```diff
@@ -153,7 +153,7 @@ func (daemon *Daemon) VolumeRm(name string) error {
 		return err
 	}
 	if err := daemon.volumes.Remove(v); err != nil {
-		if err == store.ErrVolumeInUse {
+		if err == derr.ErrorCodeVolumeInUse {
 			return fmt.Errorf("Conflict: %v", err)
```
Comment (duglin) [inline daemon/delete.go]: here too

**Comment by brahmaroutu** [comment]
Please review, PR is updated.

File: errors/volume.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		Value:          "VOLUMEEXISTS",
+		Message:        "volume already exists under %s",
+		Description:    "The specified volume already exists",
+		HTTPStatusCode: http.StatusConflict,
+	})
+
+	// ErrorCodeRemovingDirectory is generated when removing a volume that does not
+	// exist within the scope of the driver.
+	ErrorCodeRemovingDirectory = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "REMOVEVOLUMEDIRECTORY",
```
Comment (duglin) [inline errors/volume.go]: s/REMOVEVOLUMEDIRECTORY/REMOVINGDIRECTORY/

**Comment by LK4D4** [comment]
@brahmaroutu @duglin can we close this?

**Pull request closed without merging**